site/src/lib/apiClient.ts

```diff
@@ -42,17 +42,32 @@ const axiosClient = axios.create({
   withCredentials: true,
 });
 
-const handleAxiosError = (
-  error: AxiosError,
-): { error: AxiosError<{ errors: Partial<ValidationError>[] }> } => {
-  /** @todo: stop special casing unauthorized errors */
-  if (error.response?.status === 401 || error.response?.data.errors) {
-    return { error };
-  }
-  throw error;
+export type ApiClientError = AxiosError<{
+  errors?: Partial<ValidationError>[];
+}> & { parsedErrorMessage: string };
+
+const parseErrorMessageFromAxiosError = (error: ApiClientError) => {
+  const firstValidationErrorMessage = error.response?.data.errors?.find(
+    ({ msg }) => !!msg,
+  )?.msg;
+
+  return (
+    firstValidationErrorMessage ??
+    error.response?.statusText ??
+    "An unknown error occurred"
+  );
 };
 
-export type ApiClientError = AxiosError<{ errors: Partial<ValidationError>[] }>;
+const handleAxiosError = (
+  axiosError: ApiClientError,
+): { error: ApiClientError } => {
+  /** @todo: report unexpected server errors to sentry or equivalent */
+  const error = {
+    ...axiosError,
+    parsedErrorMessage: parseErrorMessageFromAxiosError(axiosError),
```

Comment:
The reason I didn't do this because it clashes with the existing `message` attribute on the `AxiosError` object, so we would be overwriting the value of that existing string. I'm unsure whether the existing `error.message` will ever be useful to us, but to avoid confusion which variables are defined by us and which are coming from `axios` I'm inclined to keep them separate. But maybe you disagree?

Reply:
Yeah, `AxiosError` inherits that attribute from `Error` and it's okay to override it. I've seen scenarios where the value is overridden when setting up axios interceptors.

Reply:
Alright, have addressed this in: https://github.com/blockprotocol/blockprotocol/pull/214/commits/69ed2a49086c3b85210a4b726a76c8879823248d